src/particle/particle/convert.py

```diff
@@ -106,7 +106,8 @@ def filter_file(fileobject: StringOrIO) -> TextIO:
 
     if not hasattr(fileobject, "read"):
         assert isinstance(fileobject, str)
-        fileobject = open(fileobject, encoding="utf-8")
+        with open(fileobject, encoding="utf-8") as f:
+            return filter_file(f)
```

Comment:
This handles the file by opening & _always_ closing it only if it's not already open. Otherwise it leaves it alone.

Reply:
Good point.